### PR TITLE
(AWS) Docs: List all AWS S3 properties from all language impl.

### PR DESCRIPTION
**Pull request by Neuw84**
As @hsiang-c  made another pull request building a table [here](https://github.com/apache/iceberg/pull/11321) I didn't want to collide. 

Fixes [List all AWS S3 properties in the docs #10674](https://github.com/apache/iceberg/issues/10674)

Therefore, I added: 

* Added Amazon MSK Connect as option.
* Added HTTP client advice when high throughput scenarios (not just tune the retries but also number of connections). 
* Added specific configs for data prefetching on EMR 7.1.0

As personal opinion, if using AWS SDKs most of the properties shouldn't be there ( there is a standard way of configure them, prioritize them, etc). However, is clear that using 3rd party libraries in different languages would require info like the tables @hsiang-c has built.

The problem with this is that different libraries will have different configs ( on the same language).  

In my personal opinion instead of dividing by language, maybe by library (but here maybe just adding a link to the corresponding doc page should be enough)? And having a separate section/table for AWS SDKs supported ones (anything using official libraries will have the same config, no matter the language)? 

Thanks!

### CODE REVIEW

File: docs/docs/aws.md [hunk 1]
```diff
@@ -717,13 +724,21 @@ install_dependencies () {
 install_dependencies $LIB_PATH $ICEBERG_MAVEN_URL $ICEBERG_VERSION "${ICEBERG_PACKAGES[@]}"
 ```
 
+For versions after 7.1.0 there is an specific config that can be used to enable data prefecth optimization. You just need to add the following property on your Spark config.
+
+```shell
+spark.sql.iceberg.data-prefetch.enabled=true
```
Comment (danielcweeks) [inline docs/docs/aws.md:730]: I don't believe this is an Iceberg property.  If this is specific to EMR, I don't believe it should be included here.

File: docs/docs/aws.md [hunk 2]
```diff
@@ -669,6 +669,13 @@ Users can use catalog properties to override the defaults. For example, to confi
 --conf spark.sql.catalog.my_catalog.http-client.apache.max-connections=5
 ```
 
+**Note that for workloads with exceptionally high throughput against tables that S3 where you will likely to increase Retries, you will also like to increase the number of connections for the HTTP client**
+
+```shell
+spark.sql.catalog.my_catalog.http-client.apache.max-connections=200
```
Comment (danielcweeks) [inline docs/docs/aws.md:675]: This doesn't look like an Iceberg setting from what I can tell.  If this is EMR specific, it should not be included here.

**Comment by danielcweeks** [comment]
@Neuw84 it looks like we're duplicating what should be EMR documentation here.  We already link off to the EMR docs, so I don't feel this is the right place for putting specific configuration info.

File: docs/docs/aws.md (hunk 2 above)
Comment (Neuw84) [reply docs/docs/aws.md:675]: It is a thing of AWS SDK and Spark ( not specifically to EMR). If you use Spark on your laptop writing to S3 and you are on this high throughput write scenario you will likely like to tune the parameter. 

I mean, we are speaking about a AWS docs in this page ( the parameter is quite specific to the S3 client of the AWS SDK).

File: docs/docs/aws.md (hunk 1 above)
Comment (Neuw84) [reply docs/docs/aws.md:730]: this is specific to EMR yes ( internal iceberg runtime), however we are on the "aws" docs page.

 I think that stating that you can add that parameter to improve the performance of AWS is good to have/know?

**Comment by Neuw84** [comment]
@danielcweeks let me know your thoughts on the comments ( agree on the specific one about EMR, although for me it does not hurt as we are on aws docs page). 

What are your thoughts about the S3 clients info?

**Pull request closed without merging**